core/metadata.js

```diff
@@ -456,3 +459,33 @@ function buildFile (filePath /*: string */, stats /*: fs.Stats */, md5sum /*: st
   if ((stats.mode & EXECUTABLE_MASK) !== 0) { doc.executable = true }
   return doc
 }
+
+const CONFLICT_REGEXP = new RegExp('-conflict-\\d{4}(?:-\\d{2}){2}T(?:\\d{2}_?){3}.\\d{3}Z')
+
+function withConflictRegExp (prefix /*: string */) /*: RegExp */ {
+  return new RegExp(`${prefix}${CONFLICT_REGEXP.source}`)
+}
+
+/**
+ *
+ * @param {Metadata} doc
+ *
+ * @returns a clone of `doc` with new `path` and `_id` properties
```

Comment:
Maybe explain that it returns a copy of the given doc with a suffix conflict, eventualy shortening the base name so the final result doesn't clash with FS limitations?